webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/uploading/UploadService.scala

```diff
@@ -185,21 +199,23 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository,
   def addUploadIdsToUnfinishedUploads(
       unfinishedUploadsWithoutIds: List[UnfinishedUpload]): Fox[List[UnfinishedUpload]] =
     for {
-      maybeUnfinishedUploads: List[Box[UnfinishedUpload]] <- Fox.sequence(
+      maybeUnfinishedUploads: List[Box[Option[UnfinishedUpload]]] <- Fox.sequence(
```

Comment:
what is the maybe encoding here? Could the listed uploads actually be finished?

Reply:
During working on this PR and checking whether the auto clean-up upon rejection (due to reserved storage overflow / too much storage after upload and in the finish upload route) I noticed that the rejected uploads were listed in the the fronted as continuable. But this is undesired in this case. This had multiple reasons:
1. Not all redis entries for the upload were clean up upon rejection
2. The postgres entry for the not finished uploaded dataset still existed

So to tackle this I tired to solve both problems by, 1. delete all redis keys, 2. delete the dataset from postgres and 3. to only send unfinished uploads which are present in the redis db.

Do you think 3rd is a little too much?

Reply:
No, I think that makes sense. The ones not in redis can’t be properly continued anyway if I’m not mistaken